Adds WC store_id to body of woopay tracker requests

FangedParakeet opened the pull request:
Fixes #10382 

#### Changes proposed in this Pull Request

<!--
Title: A descriptive, yet concise, title.
-->
This PR adds WooCommerce's `store_id` property to all WooPay tracker events sent from the plugin. 
<!--
Description: Write a brief summary about this PR. As you compose your summary, consider each of these questions and address them if appropriate. Why is this change needed? What does this change do? Were there other solutions you considered? Why did you choose to pursue this solution? Describe any trade-offs you might have had to make.
-->
Please read p7Ldg5-8Nn-p2 for more context on the motivation for this update. You can find the constant included [here](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/class-wc-tracker.php#L147) in WC's source. Also, note the similar approach implemented in Automattic/woopay#3145.
<!--
Questions for PR author:
- How can this code break?
- What are we doing to make sure this code doesn't break?
-->

<!--
Images or gifs: Include before and after screenshots or gifs/videos when it makes sense.
-->

#### Testing instructions

<!--
Testing instructions: How should this be tested and how can a reviewer test the end-user functionality? Are there known issues that you plan to address in a future PR? Are there any side effects that readers should be aware of?
-->
1. Pull the changes in this PR please.
2. To trigger some tracking events, go to Payments settings, enable or disable WooPay, and save your settings. Repeat this a few times to generate as many events as you'd like. 
3. Wait a little time (~5 mins), now visit [superset-wyeast.a8c.com/sqllab](https://superset-wyeast.a8c.com/sqllab) (you should be able to login using your Matticspace credentials) and run the query below, providing the blog ID of your test site. Hopefully, there should now be a few entries with non-null values for the `store_id` property.

```sql
SELECT
  dateymd, eventname, json_extract_scalar(eventprops, '$.store_id') AS store_id
FROM kafka_staging.etl_events
WHERE blogid = 'YOUR_BLOG_ID';
```
<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

<!--
Make sure you edit the page for the current release when adding testing instructions.
We often create a blank page ahead of time for the next release.
If this PR need not be QA tested, edit to 'QA Testing Not Applicable'
-->

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

FangedParakeet committed:
adds wc store id to body of woopay tracker requests

FangedParakeet committed:
adds changelog

FangedParakeet committed:
Merge branch 'develop' into add/10382-adds-store-id-to-tracker-request-body

botwoo commented:
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `10412` or branch name `add/10382-adds-store-id-to-tracker-request-body` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=add/10382-adds-store-id-to-tracker-request-body) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 9f90a097854904ce604c24a94fe20c25f85f9b2d
- Build time: 2025-02-21 17:11:04 UTC

_Note: the build is updated when a new commit is pushed to this PR._

timur27 reviewed (approved):
Works for me (tested with JN) for `woopay_enabled` and `woopay_disabled` events that I tested with.

<img width="867" alt="image" src="https://github.com/user-attachments/assets/1fbfc648-70ce-4dea-a86d-149f19b0aa88" />

We're good to go, the only thing would be to make the constant resolvable in tests which are now failing.

FangedParakeet committed:
Merge branch 'develop' into add/10382-adds-store-id-to-tracker-request-body

FangedParakeet committed:
ensures wc constant is defined, adds unit test with mocked method

FangedParakeet committed:
Merge branch 'add/10382-adds-store-id-to-tracker-request-body' of github.com:Automattic/woocommerce-payments into add/10382-adds-store-id-to-tracker-request-body

FangedParakeet committed:
Merge branch 'develop' into add/10382-adds-store-id-to-tracker-request-body

FangedParakeet committed:
fixes linter errors

FangedParakeet commented:
@timur27, it appears refreshingly that the tests were actually failing with good reason. The `WC_Install::STORE_ID_OPTION` was only introduced in WC 8.4 and since our minimum WC version is 7.6, I added a few changes to support some backwards compatibility here. 

* Added a check that the WC constant is defined.
* Moved logic to a separate function.
* Mocked out above function and added parameter check to existing unit tests.

If you wouldn't mind giving this another quick gander, lemme know if this still all looks tippity top. Cheers!

timur27 reviewed (approved):
Now I know why it worked on my JN site :) Thanks for adding this! 

This makes sense, although the fact of us mocking the system under test in `tests/unit/test-class-woopay-tracker.php` is something we could consider changing. I don't think there are better workarounds, though. We could start mocking the object specifically in the tests which retrieve the option while keeping the main system under test real, but I guess all the tests in this suite retrieve this option.
